### PR TITLE
Logged EtcdKeyNotFound exception from rpc.py

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Instead of the exception, EtcdKeyNotFound is logged successfully
tendrl-bug-id:Tendrl/ceph_integration#14

Signed-off-by: root <gshanmug@redhat.com>

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
@r0h4n @shtripat @nnDarshan please review it

**Review by nnDarshan — COMMENTED** [review]
looks good to me

File: tendrl/ceph_integration/manager/rpc.py [hunk 1]
```diff
@@ -162,6 +162,9 @@ def _run(self):
             try:
                 LOG.info("%s run..." % self.__class__.__name__)
                 self._server.run()
+            except etcd.EtcdKeyNotFound as ex:
+                LOG.debug("Given key is not present in etcd . %s", ex)
```
Comment (shtripat) [inline tendrl/ceph_integration/manager/rpc.py:168]: I would prefer printing the key name in log message

**Review by shtripat — COMMENTED** [review]
Looks good to me

File: tendrl/ceph_integration/manager/rpc.py [hunk 2]
```diff
@@ -163,6 +163,9 @@ def _run(self):
             try:
                 LOG.info("%s run..." % self.__class__.__name__)
                 self._server.run()
+            except etcd.EtcdKeyNotFound as ex:
```
Comment (r0h4n) [inline tendrl/ceph_integration/manager/rpc.py:167]: Needs negative test case to assert if the etcd.EtcdKeyNotFound is caught

**Review by r0h4n — CHANGES_REQUESTED** [review]

**Comment by GowthamShanmugam** [comment]
@r0h4n please review it

File: tendrl/ceph_integration/manager/rpc.py [hunk 3]
```diff
@@ -20,13 +20,14 @@
 
 class EtcdRPC(object):
 
-    def __init__(self, manager):
+    def __init__(self, manager, Etcdthread):
```
Comment (r0h4n) [inline tendrl/ceph_integration/manager/rpc.py:23]: why is this change required here, is the same required in tendrl/gluster_integration/manager?

**Review by r0h4n — CHANGES_REQUESTED** [review]
Actual PR looks good, but needinfo about the unrelated changes

File: tendrl/ceph_integration/manager/rpc.py (hunk 3 above)
Comment (GowthamShanmugam) [reply tendrl/ceph_integration/manager/rpc.py:23]: EtcdRPC  have infinite loop. It will never stop. We have this check in other repos also.

**Review by r0h4n — APPROVED** [review]

File: tendrl/ceph_integration/manager/rpc.py (hunk 3 above)
Comment (r0h4n) [reply tendrl/ceph_integration/manager/rpc.py:23]: @GowthamShanmugam Please create issues on node-agent, gluster-integration, performance-monitoring, alerting

Comment (GowthamShanmugam) [reply tendrl/ceph_integration/manager/rpc.py:23]: ok i will create

**Pull request merged**